device/cuda/include/traccc/cuda/fitting/fitting_algorithm.hpp

```diff
@@ -0,0 +1,54 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "traccc/edm/track_candidate.hpp"
+#include "traccc/edm/track_state.hpp"
+#include "traccc/utils/algorithm.hpp"
+#include "traccc/utils/memory_resource.hpp"
+
+// VecMem include(s).
+#include <vecmem/utils/copy.hpp>
+#include <vecmem/utils/cuda/copy.hpp>
+
+// traccc library include(s).
+#include "traccc/utils/memory_resource.hpp"
+
+namespace traccc::cuda {
+
+/// Fitting algorithm for a set of tracks
+template <typename fitter_t, typename host_detector_t>
```

Comment:
Hmm. you are right. I don't want `host_detector_t` or `detector_view` as a template parameter. 

However, `fitter_t` only contains the device detector type, and we need host detector objects as an input. I know it could sound silly - Can we create a device detector from the host side and pass it to the device?  If that's possible, we can remove the host detector type input

Reply:
OK. I think I can (hopefully) make `fitter_t` figure out the `detector_view` object

Reply:
The template parameter of `host_detector_t` is removed now